dracut/repartition/endless-repartition.sh

```diff
@@ -151,9 +144,17 @@ fi
 lastpartno=$(echo "$parts" | sed -n -e '$ s/[^:]*\([0-9]\) :.*$/\1/p')
 
 if [ $lastpartno -eq $swap_partno ]; then
-  # already have an extra partition, perhaps we were halfway through creating
-  # a swap partition but didn't finish. Remove it to try again below.
+  # We previously created an extra partition for swap.
+  # Remove it, now that we are using zram instead of disk swap.
+  # Keep in mind, though, that this code path currently only runs
+  # if we had not yet completed the repartitioning, for example
+  # in the rare case that we were halfway through creating a swap
+  # partition on a previous version of the OS, but did not finish
+  # before upgrading the OS.
   parts=$(echo "$parts" | sed '$d')
+  # Schedule a re-run of the filesystem resizing service
+  # to reclaim the extra disk space.
+  rm -f /var/eos-booted
```

Comment:
This script runs from the initramfs: is the filesystem on this drive, and hence `/var`, really mounted at this point? This comment from later in the script suggests not:

```
 # Randomize root filesystem UUID
 # uninit_bg functionality prevents us from doing this later when the
 # filesystem is mounted.
```

I think we'll need to do this from the booted system, in whatever future job readds the GUID:55 marker to trigger this script running on next boot.

Reply:
Agreed